pre-commit: integrate with cndp

maryamtahhan opened the pull request:
Install pre-commit for CNDP + its configuration.
Remove old githook.

maryamtahhan commented on doc/api/doxy-api-index.md:
```diff
@@ -1,5 +1,4 @@
-API {#index}
-===
+# API {#index}
```
Restore this file and add it to the ignore list

maryamtahhan commented on examples/dlb_test/README.md:
```diff
@@ -17,15 +17,16 @@ The tarball contains a DLB driver and libdlb (user space library).
+Create a directory named dlb and copy all the header files from
+$BASE_DLB_DIR/libdlb\*.h into /usr/local/include/dlb CNDP will then link the
+library to the app if added as a dependency in the example's meson file deps +=
+\[libdlb\]
```
Double check this

maryamtahhan commented on lang/rs/wireguard/patch/0002-Remove-LD_PRELOAD-from-cndprustwg-script.patch:
```diff
@@ -18,9 +18,9 @@ index f390b63..d23f822 100644
 --- a/src/platform/linux/cndp/README.md
```
Restore patch files

maryamtahhan commented on lang/rs/wireguard/patch/0003-Update-latest-crypto-library-version.patch:
```diff
@@ -32,15 +32,15 @@ index 2941071..51cd887 100644
  dependencies = [
```
restore

maryamtahhan commented on test/common/test.h:
```diff
@@ -182,9 +182,6 @@ void add_test_command(struct test_command *t);
-    {                                                \
-        add_test_command(&test_struct_##cmd);        \
-    }
+    CNE_INIT(test_register_##cmd) { add_test_command(&test_struct_##cmd); }
```
Double check clang formatting

maryamtahhan commented on usrtools/txgen/README.md:
```diff
@@ -1,21 +1,22 @@
 # TXGen - Traffic Generator powered by CNDP
 
-=====================================================
+\=====================================================
```
```suggestion
---
```

maryamtahhan commented:
Sorry for the noise - it took a while to find the right recipe of clang versions between pre-commit hooks and the github action

maryamtahhan commented:
:) Markdown linter disconnect now :(

KeithWiles reviewed (approved):
Thanks Maryam, that is a lot of work and it looks good to me.

KeithWiles commented:
Should this get merged now or will you be fixing the Lint Code Base issues?

maryamtahhan commented:
I will fix them :)

maryamtahhan commented:
Linting issues are now fixed. I just rebased there and kicked off the github actions again. As soon as they complete we should be good. 

Sorry again for all the noise :( I couldn't reproduce the issues locally

KeithWiles commented:
No problem it is a lot of work, Thanks
